jxl/src/render/stages/chroma_upsample.rs

```diff
@@ -25,6 +26,59 @@ impl std::fmt::Display for HorizontalChromaUpsample {
     }
 }
 
+// SIMD horizontal chroma upsampling
+simd_function!(
+    hchroma_upsample_simd_dispatch,
+    d: D,
+    fn hchroma_upsample_simd(input: &[f32], output: &mut [f32], xsize: usize) {
+        let simd_width = D::F32Vec::LEN;
+
+        // Precompute constants
+        let c025 = D::F32Vec::splat(d, 0.25);
+        let c075 = D::F32Vec::splat(d, 0.75);
+
+        // Temporary buffer for storing SIMD results before interleaving
+        let mut left_buf = [0.0f32; 16];
+        let mut right_buf = [0.0f32; 16];
+
+        // SIMD loop
+        let mut x = 0;
+        while x + simd_width <= xsize {
+            // Load: prev, cur, next (with +1 offset for border)
+            let prev_vec = D::F32Vec::load(d, &input[x..]);
```

Comment:
We might want to use `.windows().step_by()` for the input and `.chunks_exact_mut()` for the output instead.